.azure-pipelines/bazel.yml

```diff
@@ -55,7 +55,6 @@ steps:
       BAZEL_REMOTE_INSTANCE: projects/envoy-ci/instances/default_instance
     ${{ if eq(parameters.rbe, false) }}:
       BAZEL_BUILD_EXTRA_OPTIONS: "${{ parameters.bazelBuildExtraOptions }}"
-      BAZEL_REMOTE_CACHE: $(LocalBuildCache)
```

Comment:
@mattklein123 i forgot about this bit - we need to think of a better way for this (#24001) or at very least automate/doc it as part of release branching

failure to do this means the release branch foos the main branch caches and ci becomes uber flakey